// ngx-emoj/esm2015/lib/components/category.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, Input, Output, EventEmitter } from '@angular/core';
import { DomSanitizer } from '@angular/platform-browser';
export class NgxEmojCategoryComponent {
    /**
     * @param {?} sanitizer
     */
    constructor(sanitizer) {
        this.sanitizer = sanitizer;
        this.onselect = new EventEmitter;
    }
    /**
     * @return {?}
     */
    selectCategory() {
        this.onselect.emit({ name: this.categoryName, icon: this.categoryIcon });
    }
}
NgxEmojCategoryComponent.decorators = [
    { type: Component, args: [{
                selector: 'ngx-emoj-category',
                template: `
    <button (click)="selectCategory()" class="ngx-emoji-category-btn"
    [ngStyle]="{'color': categoryIconColor,
                'border-width': activeIndicatorHeight,
                'border-color': (active) ? activeIndicatorColor : 'transparent'}"
                [innerHTML]="sanitizer.bypassSecurityTrustHtml(
                  categoryIcon(((active) ? martCategoryColorActive : martCategoryColor), martCategoryFontSize, martCategoryFontSize))">
    </button>
  `,
                styles: [`
  .ngx-emoji-category-btn
  {
    background: transparent;
    padding: 15px 10% 10px 10%;
    border: none;
    outline: none;
    border-bottom: 2px solid transparent;
  }
  `]
            }] }
];
/** @nocollapse */
NgxEmojCategoryComponent.ctorParameters = () => [
    { type: DomSanitizer }
];
NgxEmojCategoryComponent.propDecorators = {
    categoryIcon: [{ type: Input }],
    categoryName: [{ type: Input }],
    categoryIconColor: [{ type: Input }],
    onselect: [{ type: Output }],
    active: [{ type: Input }],
    activeIndicatorColor: [{ type: Input }],
    activeIndicatorHeight: [{ type: Input }],
    martCategoryFontSize: [{ type: Input }],
    martCategoryColor: [{ type: Input }],
    martCategoryColorActive: [{ type: Input }]
};
if (false) {
    /** @type {?} */
    NgxEmojCategoryComponent.prototype.categoryIcon;
    /** @type {?} */
    NgxEmojCategoryComponent.prototype.categoryName;
    /** @type {?} */
    NgxEmojCategoryComponent.prototype.categoryIconColor;
    /** @type {?} */
    NgxEmojCategoryComponent.prototype.onselect;
    /** @type {?} */
    NgxEmojCategoryComponent.prototype.active;
    /** @type {?} */
    NgxEmojCategoryComponent.prototype.activeIndicatorColor;
    /** @type {?} */
    NgxEmojCategoryComponent.prototype.activeIndicatorHeight;
    /** @type {?} */
    NgxEmojCategoryComponent.prototype.martCategoryFontSize;
    /** @type {?} */
    NgxEmojCategoryComponent.prototype.martCategoryColor;
    /** @type {?} */
    NgxEmojCategoryComponent.prototype.martCategoryColorActive;
    /** @type {?} */
    NgxEmojCategoryComponent.prototype.sanitizer;
}
