// ngx-emoj/esm2015/lib/ngx-emoj.module.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { NgxEmojComponent } from './ngx-emoj.component';
import { NgxEmojHeaderComponent } from './components/header.component';
import { NgxEmojFooterComponent } from './components/footer.component';
import { NgxEmojCategoryComponent } from './components/category.component';
import { NgxEmojCategoryContentComponent } from './components/category-content.component';
export class NgxEmojModule {
}
NgxEmojModule.decorators = [
    { type: NgModule, args: [{
                declarations: [
                    NgxEmojComponent,
                    NgxEmojHeaderComponent,
                    NgxEmojCategoryComponent,
                    NgxEmojCategoryContentComponent,
                    NgxEmojFooterComponent
                ],
                imports: [
                    CommonModule
                ],
                exports: [NgxEmojComponent,
                    NgxEmojHeaderComponent,
                    NgxEmojCategoryComponent,
                    NgxEmojCategoryContentComponent,
                    NgxEmojFooterComponent],
                providers: [],
                entryComponents: [NgxEmojComponent]
            },] }
];
