// ngx-emoj/esm5/lib/components/category.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, Input, Output, EventEmitter } from '@angular/core';
import { DomSanitizer } from '@angular/platform-browser';
var NgxEmojCategoryComponent = /** @class */ (function () {
    function NgxEmojCategoryComponent(sanitizer) {
        this.sanitizer = sanitizer;
        this.onselect = new EventEmitter;
    }
    /**
     * @return {?}
     */
    NgxEmojCategoryComponent.prototype.selectCategory = /**
     * @return {?}
     */
    function () {
        this.onselect.emit({ name: this.categoryName, icon: this.categoryIcon });
    };
    NgxEmojCategoryComponent.decorators = [
        { type: Component, args: [{
                    selector: 'ngx-emoj-category',
                    template: "\n    <button (click)=\"selectCategory()\" class=\"ngx-emoji-category-btn\"\n    [ngStyle]=\"{'color': categoryIconColor,\n                'border-width': activeIndicatorHeight,\n                'border-color': (active) ? activeIndicatorColor : 'transparent'}\"\n                [innerHTML]=\"sanitizer.bypassSecurityTrustHtml(\n                  categoryIcon(((active) ? martCategoryColorActive : martCategoryColor), martCategoryFontSize, martCategoryFontSize))\">\n    </button>\n  ",
                    styles: ["\n  .ngx-emoji-category-btn\n  {\n    background: transparent;\n    padding: 15px 10% 10px 10%;\n    border: none;\n    outline: none;\n    border-bottom: 2px solid transparent;\n  }\n  "]
                }] }
    ];
    /** @nocollapse */
    NgxEmojCategoryComponent.ctorParameters = function () { return [
        { type: DomSanitizer }
    ]; };
    NgxEmojCategoryComponent.propDecorators = {
        categoryIcon: [{ type: Input }],
        categoryName: [{ type: Input }],
        categoryIconColor: [{ type: Input }],
        onselect: [{ type: Output }],
        active: [{ type: Input }],
        activeIndicatorColor: [{ type: Input }],
        activeIndicatorHeight: [{ type: Input }],
        martCategoryFontSize: [{ type: Input }],
        martCategoryColor: [{ type: Input }],
        martCategoryColorActive: [{ type: Input }]
    };
    return NgxEmojCategoryComponent;
}());
export { NgxEmojCategoryComponent };
if (false) {
    /** @type {?} */
    NgxEmojCategoryComponent.prototype.categoryIcon;
    /** @type {?} */
    NgxEmojCategoryComponent.prototype.categoryName;
    /** @type {?} */
    NgxEmojCategoryComponent.prototype.categoryIconColor;
    /** @type {?} */
    NgxEmojCategoryComponent.prototype.onselect;
    /** @type {?} */
    NgxEmojCategoryComponent.prototype.active;
    /** @type {?} */
    NgxEmojCategoryComponent.prototype.activeIndicatorColor;
    /** @type {?} */
    NgxEmojCategoryComponent.prototype.activeIndicatorHeight;
    /** @type {?} */
    NgxEmojCategoryComponent.prototype.martCategoryFontSize;
    /** @type {?} */
    NgxEmojCategoryComponent.prototype.martCategoryColor;
    /** @type {?} */
    NgxEmojCategoryComponent.prototype.martCategoryColorActive;
    /** @type {?} */
    NgxEmojCategoryComponent.prototype.sanitizer;
}
