// ngx-emoj/esm5/lib/components/footer.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, Input, Output, EventEmitter } from '@angular/core';
import { DomSanitizer } from '@angular/platform-browser';
var NgxEmojFooterComponent = /** @class */ (function () {
    function NgxEmojFooterComponent(sanitizer) {
        this.sanitizer = sanitizer;
        this.onemochange = new EventEmitter();
        this.onchardelete = new EventEmitter();
    }
    /**
     * @return {?}
     */
    NgxEmojFooterComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        this.activeEmo = this.defaultActiveEmo;
    };
    /**
     * @param {?} e
     * @return {?}
     */
    NgxEmojFooterComponent.prototype.onEmoSelect = /**
     * @param {?} e
     * @return {?}
     */
    function (e) {
        this.activeEmo = e.name;
        this.onemochange.emit({ name: e.name });
    };
    /**
     * @param {?} e
     * @return {?}
     */
    NgxEmojFooterComponent.prototype.deleteChar = /**
     * @param {?} e
     * @return {?}
     */
    function (e) {
        this.onchardelete.emit({ deleteChar: true });
    };
    /**
     * @param {?} fill
     * @param {?} width
     * @param {?} height
     * @return {?}
     */
    NgxEmojFooterComponent.prototype.delButton = /**
     * @param {?} fill
     * @param {?} width
     * @param {?} height
     * @return {?}
     */
    function (fill, width, height) {
        return "<svg version=\"1.1\" id=\"Capa_1\" xmlns=\"http://www.w3.org/2000/svg\" xmlns:xlink=\"http://www.w3.org/1999/xlink\" x=\"0px\" y=\"0px\" viewBox=\"0 0 511.374 511.374\" width=\"" + width + "\" height=\"" + height + "\" fill=\"" + fill + "\" style=\"enable-background:new 0 0 511.374 511.374\" xml:space=\"preserve\"><g><g><path d=\"M433.134,47.687h-213.28c-37.109,0.21-72.235,16.778-96,45.28l-1.44,1.6l-115.2,140.8c-9.619,11.779-9.619,28.701,0,40.48l115.36,141.92c23.6,28.794,58.771,45.618,96,45.92h216.16c41.414-2.536,74.363-35.691,76.64-77.12c0-0.96,0-1.92,0-2.88v-257.12C510.781,83.499,476.196,48.631,433.134,47.687z M447.374,382.567c-0.428,9.583-8.327,17.13-17.92,17.12h-208c-19.662,0.779-38.49-7.979-50.56-23.52l-98.24-120.48l97.92-119.68c11.851-15.727,30.553-24.78,50.24-24.32h209.6c11.04,0,16,6.4,16.96,16V382.567z\"/></g></g><g><g><path d=\"M373.934,296.967l-11.52-11.52c-12.504-12.504-32.776-12.504-45.28,0s-12.504,32.776,0,45.28l11.52,11.52c12.504,12.504,32.776,12.504,45.28,0S386.438,309.471,373.934,296.967z\"/></g></g><g><g><path d=\"M373.934,169.127c-12.504-12.504-32.776-12.504-45.28,0h0.16l-41.44,41.28l-41.44-41.44c-12.504-12.504-32.776-12.504-45.28,0s-12.504,32.776,0,45.28l41.44,41.44l-41.44,41.44c-12.504,12.504-12.504,32.776,0,45.28s32.776,12.504,45.28,0l128-128C386.438,201.903,386.438,181.631,373.934,169.127z\"/></g></g><g></g><g></g><g></g><g></g><g></g><g></g><g></g><g></g><g></g><g></g><g></g><g></g><g></g><g></g><g></g></svg>";
    };
    NgxEmojFooterComponent.decorators = [
        { type: Component, args: [{
                    selector: 'ngx-emoj-footer',
                    template: "\n    <div\n      class=\"ngx-emoj-footer\"\n      [class.unveal]=\"hideFooter\"\n      [class.reveal]=\"!hideFooter\"\n      [ngStyle]=\"{'background-color': footerBG,\n                  'color': footerFG,\n                  'font-size': footerFontSize,\n                  'padding': footerPadding.y+' '+footerPadding.x}\">\n\n                  <div class=\"l\">\n                        <button\n                        class=\"emos-btn\"\n                *ngFor=\"let e of emos\" [innerHTML]=\"sanitizer.bypassSecurityTrustHtml(\n                          e.icon[0](((e.name == activeEmo ) ? martCategoryColorActive : martCategoryColor),\n                                       martCategoryFontSize, martCategoryFontSize))\">\n                        </button>\n                  </div>\n                  <div class=\"r\">\n                      <button class=\"emos-btn\"\n                          (click)=\"deleteChar($event)\"\n                          [ngStyle]=\"{'color': martCategoryColor}\"\n                          [innerHTML]=\"sanitizer.bypassSecurityTrustHtml(\n                            delButton(martCategoryColor, martCategoryFontSize, martCategoryFontSize))\">\n                      </button>\n                  </div>\n    </div>\n  ",
                    styles: ["\n   .ngx-emoj-footer\n  {\n    display: flex;\n    justify-content: center;\n    flex-wrap: wrap;\n    align-items: center;\n    width: 100%;\n    box-sizing: border-box;\n    position: absolute;\n    bottom: 0;\n    left: 0;\n\n  }\n  .ngx-emoj-footer .l\n  {\n    display: flex;\n    justify-content: center;\n    align-items: center;\n    width: 85%;\n  }\n  .ngx-emoj-footer .r\n  {\n    display: flex;\n    justify-content: center;\n    align-items: center;\n    width: 15%;\n  }\n\n  .emos-btn\n  {\n    background: transparent;\n    padding: 15px 10% 10px 10%;\n    border: none;\n    outline: none;\n    padding-left: 10%;\n    padding-right: 10%;\n    border: none;\n  }\n\n\n  .reveal\n  {\n    animation-name: reveal;\n    animation-duration: .3s;\n    animation-fill-mode: forwards;\n  }\n\n\n  .unveal\n  {\n    animation-name: unveal;\n    animation-duration: .5s;\n    animation-fill-mode: forwards;\n  }\n\n  @keyframes unveal\n  {\n    from\n    {\n      opacity: 1;\n      bottom: 0;\n    }\n\n    to\n    {\n      opacity: 0;\n      bottom: -150px;\n    }\n  }\n\n  @keyframes reveal\n  {\n\n    from\n    {\n      opacity: 0;\n      bottom: -150px;\n    }\n\n    to\n    {\n      opacity: 1;\n      bottom: 0;\n    }\n  }\n\n  "]
                }] }
    ];
    /** @nocollapse */
    NgxEmojFooterComponent.ctorParameters = function () { return [
        { type: DomSanitizer }
    ]; };
    NgxEmojFooterComponent.propDecorators = {
        footerBG: [{ type: Input }],
        footerFG: [{ type: Input }],
        footerFontSize: [{ type: Input }],
        footerPadding: [{ type: Input }],
        emos: [{ type: Input }],
        activeIndicatorColor: [{ type: Input }],
        activeIndicatorHeight: [{ type: Input }],
        defaultActiveEmo: [{ type: Input }],
        onemochange: [{ type: Output }],
        martCategoryFontSize: [{ type: Input }],
        martCategoryColor: [{ type: Input }],
        martCategoryColorActive: [{ type: Input }],
        onchardelete: [{ type: Output }],
        hideFooter: [{ type: Input }]
    };
    return NgxEmojFooterComponent;
}());
export { NgxEmojFooterComponent };
if (false) {
    /** @type {?} */
    NgxEmojFooterComponent.prototype.footerBG;
    /** @type {?} */
    NgxEmojFooterComponent.prototype.footerFG;
    /** @type {?} */
    NgxEmojFooterComponent.prototype.footerFontSize;
    /** @type {?} */
    NgxEmojFooterComponent.prototype.footerPadding;
    /** @type {?} */
    NgxEmojFooterComponent.prototype.emos;
    /** @type {?} */
    NgxEmojFooterComponent.prototype.activeIndicatorColor;
    /** @type {?} */
    NgxEmojFooterComponent.prototype.activeIndicatorHeight;
    /** @type {?} */
    NgxEmojFooterComponent.prototype.defaultActiveEmo;
    /** @type {?} */
    NgxEmojFooterComponent.prototype.onemochange;
    /** @type {?} */
    NgxEmojFooterComponent.prototype.martCategoryFontSize;
    /** @type {?} */
    NgxEmojFooterComponent.prototype.martCategoryColor;
    /** @type {?} */
    NgxEmojFooterComponent.prototype.martCategoryColorActive;
    /** @type {?} */
    NgxEmojFooterComponent.prototype.onchardelete;
    /** @type {?} */
    NgxEmojFooterComponent.prototype.hideFooter;
    /** @type {?} */
    NgxEmojFooterComponent.prototype.activeEmo;
    /** @type {?} */
    NgxEmojFooterComponent.prototype.sanitizer;
}
