// ngx-emoj/esm5/lib/components/header.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, Input, Output, EventEmitter } from '@angular/core';
var NgxEmojHeaderComponent = /** @class */ (function () {
    function NgxEmojHeaderComponent() {
        this.oncategorychange = new EventEmitter();
    }
    /**
     * @return {?}
     */
    NgxEmojHeaderComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        if (!this.activeCategory) {
            this.activeCategory = this.defaultActiveCategory;
        }
    };
    /**
     * @param {?} e
     * @return {?}
     */
    NgxEmojHeaderComponent.prototype.onCategorySelect = /**
     * @param {?} e
     * @return {?}
     */
    function (e) {
        this.activeCategory = e.name;
        this.oncategorychange.emit({ name: e.name });
    };
    NgxEmojHeaderComponent.decorators = [
        { type: Component, args: [{
                    selector: 'ngx-emoj-header',
                    template: "\n    <div\n      class=\"ngx-emoj-header\"\n      [ngStyle]=\"{'background-color': headerBG,\n                  'color': headerFG,\n                  'font-size': headerFontSize,\n                  'padding': headerPadding.y+' '+headerPadding.x}\">\n\n                  <ngx-emoj-category *ngFor=\"let c of emojiCategories\"\n                   [categoryIcon]=\"c.icon[0]\"\n                   [categoryIconColor]=\"'white'\"\n                   [categoryName]=\"c.name\"\n                   [martCategoryFontSize]=\"martCategoryFontSize\"\n                   [martCategoryColor]=\"martCategoryColor\"\n                   [martCategoryColorActive]=\"martCategoryColorActive\"\n                   [activeIndicatorColor]=\"activeIndicatorColor\"\n                   [activeIndicatorHeight]=\"activeIndicatorHeight\"\n                   [active]=\"activeCategory === c.name\"\n                   (onselect)=\"onCategorySelect($event)\">\n                  </ngx-emoj-category>\n    </div>\n  ",
                    styles: ["\n\n  .ngx-emoj-header\n  {\n    display: flex;\n    justify-content: space-around;\n    flex-wrap: wrap;\n    align-items: center;\n    width: 100%;\n    box-sizing: border-box;\n  }\n\n\n  "]
                }] }
    ];
    /** @nocollapse */
    NgxEmojHeaderComponent.ctorParameters = function () { return []; };
    NgxEmojHeaderComponent.propDecorators = {
        headerBG: [{ type: Input }],
        headerFG: [{ type: Input }],
        headerFontSize: [{ type: Input }],
        headerPadding: [{ type: Input }],
        emojiCategories: [{ type: Input }],
        activeIndicatorColor: [{ type: Input }],
        activeIndicatorHeight: [{ type: Input }],
        defaultActiveCategory: [{ type: Input }],
        oncategorychange: [{ type: Output }],
        martCategoryFontSize: [{ type: Input }],
        martCategoryColor: [{ type: Input }],
        martCategoryColorActive: [{ type: Input }],
        activeCategory: [{ type: Input }]
    };
    return NgxEmojHeaderComponent;
}());
export { NgxEmojHeaderComponent };
if (false) {
    /** @type {?} */
    NgxEmojHeaderComponent.prototype.headerBG;
    /** @type {?} */
    NgxEmojHeaderComponent.prototype.headerFG;
    /** @type {?} */
    NgxEmojHeaderComponent.prototype.headerFontSize;
    /** @type {?} */
    NgxEmojHeaderComponent.prototype.headerPadding;
    /** @type {?} */
    NgxEmojHeaderComponent.prototype.emojiCategories;
    /** @type {?} */
    NgxEmojHeaderComponent.prototype.activeIndicatorColor;
    /** @type {?} */
    NgxEmojHeaderComponent.prototype.activeIndicatorHeight;
    /** @type {?} */
    NgxEmojHeaderComponent.prototype.defaultActiveCategory;
    /** @type {?} */
    NgxEmojHeaderComponent.prototype.oncategorychange;
    /** @type {?} */
    NgxEmojHeaderComponent.prototype.martCategoryFontSize;
    /** @type {?} */
    NgxEmojHeaderComponent.prototype.martCategoryColor;
    /** @type {?} */
    NgxEmojHeaderComponent.prototype.martCategoryColorActive;
    /** @type {?} */
    NgxEmojHeaderComponent.prototype.activeCategory;
}
